tajo-core/src/main/java/org/apache/tajo/master/exec/QueryExecutor.java

```diff
@@ -260,6 +260,7 @@ public void execSimpleQuery(QueryContext queryContext, Session session, String q
     if (plan.getRootBlock().hasNode(NodeType.LIMIT)) {
       LimitNode limitNode = plan.getRootBlock().getNode(NodeType.LIMIT);
       maxRow = (int) limitNode.getFetchFirstNum();
+      scanNode.setLimit(maxRow);
```

Comment:
It seems that this change makes NonForwardQueryResultFileScanner.maxRow unnecessary.